### ISSUE TITLE
access permissions do not include g+r

### ISSUE THREAD

**Issue by mfrasca** [Homebrew/homebrew#40075]
I've been installing stuff on my boss' computer, she made a user for myself and gave me administration rights. then I needed do some fine-tuning from her terminal and to do so I needed correcting ownership of things in `/Library/Caches/Homebrew/Formula/` and in `/usr/local/`.

`brew` would create files giving ownership to the user running brew and setting permissions as `u+rwX,go+rX`, barring out joint administration.

my workaround was adding permission `g+w`.

**Comment by mikemcquaid** [comment]
That's due to your default `umask`. That workaround is expected, I'm afraid.

**Comment by mfrasca** [comment]
but why doesn't brew install stuff as being root? then it would work not depending on my user settings.

**Comment by mikemcquaid** [comment]
Homebrew doesn't install as root for security reasons so you don't need to trust Homebrew or any of the build scripts with root privileges.

**Comment by mfrasca** [comment]
ok, but the way it works now, it depends on user setting whether or not it will work for every user. my impression is that it would be safer to alter permissions of everything one installs (regardless their umask), to ug+rwX,o_rX. 
as of now, if I want to break homebrew for other users, I just set my umask to something less permissive and start installing things with it.

**Comment by mikemcquaid** [comment]
This is the way pretty much all Unix commands behave.

**Comment by mfrasca** [comment]
I don't know which 'Unix' you refer to, I know Linux, and as said, on my Linux machines (ubuntu, debian, fedora), my `/usr/local/` does not hold files belonging to me: all belong to root.
but I'm not making it a point of theory: my point is that it is in practice easy for one to forget setting the umask, or to open g+w, with the consequence that homebrew will make things (slightly more) difficult to the next user wanting to install anything (in my case, I had to complete the job I had initiated under a different account). On computers which are single user, it will work fine, obviously.

why don't you just make Homebrew check that umask is set correctly for multi-user machines?
